libraries/chain/include/graphene/chain/protocol/sidechain_transaction.hpp

```diff
@@ -1,60 +1,72 @@
+      std::string transaction;
+      std::vector<son_id_type> signers;
 
       account_id_type fee_payer()const { return payer; }
-      void            validate()const {}
       share_type      calculate_fee(const fee_parameters_type& k)const { return 0; }
    };
 
-   struct bitcoin_transaction_sign_operation : public base_operation
+   struct sidechain_transaction_sign_operation : public base_operation
    {
       struct fee_parameters_type { uint64_t fee = 0; };
 
-      asset                         fee;
-      account_id_type               payer;
-      proposal_id_type              proposal_id;
-      std::vector<peerplays_sidechain::bytes> signatures;
+      asset fee;
+      account_id_type payer;
+
+      sidechain_transaction_id_type sidechain_transaction_id;
+      std::string transaction;
+      bool complete;
 
       account_id_type   fee_payer()const { return payer; }
-      void              validate()const {}
       share_type        calculate_fee( const fee_parameters_type& k )const { return 0; }
    };
 
-   struct bitcoin_send_transaction_process_operation : public base_operation
+   struct sidechain_transaction_send_operation : public base_operation
    {
       struct fee_parameters_type { uint64_t fee = 0; };
 
-      asset                         fee;
-      account_id_type               payer;
+      asset fee;
+      account_id_type payer;
 
-      bitcoin_transaction_id_type   bitcoin_transaction_id;
+      sidechain_transaction_id_type sidechain_transaction_id;
 
       account_id_type   fee_payer()const { return payer; }
-      void              validate()const {}
       share_type        calculate_fee( const fee_parameters_type& k )const { return 0; }
    };
 
 } } // graphene::chain
 
-FC_REFLECT( graphene::chain::bitcoin_transaction_send_operation::fee_parameters_type, (fee) )
-FC_REFLECT( graphene::chain::bitcoin_transaction_send_operation, (fee)(payer)(signatures) )
-
-FC_REFLECT( graphene::chain::bitcoin_transaction_sign_operation::fee_parameters_type, (fee) )
-FC_REFLECT( graphene::chain::bitcoin_transaction_sign_operation, (fee)(payer)(proposal_id)(signatures) )
-
-FC_REFLECT( graphene::chain::bitcoin_send_transaction_process_operation::fee_parameters_type, (fee) )
-FC_REFLECT( graphene::chain::bitcoin_send_transaction_process_operation, (fee)(payer)(bitcoin_transaction_id) )
\ No newline at end of file
+FC_REFLECT( graphene::chain::sidechain_transaction_create_operation::fee_parameters_type, (fee) )
+FC_REFLECT( graphene::chain::sidechain_transaction_create_operation, (fee)(payer)
+        (sidechain)
+        (son_wallet_id)
+        (son_wallet_deposit_id)
+        (son_wallet_withdraw_id)
+        (transaction)
+        (signers) )
+
+FC_REFLECT( graphene::chain::sidechain_transaction_sign_operation::fee_parameters_type, (fee) )
+FC_REFLECT( graphene::chain::sidechain_transaction_sign_operation, (fee)(payer)
+        (sidechain_transaction_id)
+        (transaction) )
```

Comment:
Shoud we serialize `complete` field too?

Reply:
Done